base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,1391 @@
+            if uom:
+                return uom
+        if product:
+            return product.uom_id
+        chatter_msg.append(
+            _(
+                "<p>Odoo couldn't find any unit of measure corresponding to the "
+                "following information extracted from the business document:</p>"
+                "<ul><li>UNECE code: %(code)s</li>"
+                "<li>Name of the unit of measure: %(name)s</li></ul>"
+                "<p>So the unit of measure 'Unit(s)' has been used. <em>You may "
+                "have to change it manually.</em></p>",
+                code=uom_dict.get("unece_code"),
+                name=uom_dict.get("name"),
+            )
+        )
+        return self.env.ref("uom.product_uom_unit")
+
+    @api.model
+    def _match_taxes(
+        self, taxes_list, chatter_msg, type_tax_use="purchase", price_include=False
+    ):
+        """taxes_list must be a list of tax_dict"""
+        taxes_recordset = self.env["account.tax"].browse(False)
+        for tax_dict in taxes_list:
+            taxes_recordset += self._match_tax(
+                tax_dict,
+                chatter_msg,
+                type_tax_use=type_tax_use,
+                price_include=price_include,
+            )
+        return taxes_recordset
+
+    @api.model
+    def _prepare_match_tax_domain(
+        self, tax_dict, type_tax_use="purchase", price_include=False
+    ):
+        ato = self.env["account.tax"]
+        company_id = self._context.get("force_company") or self.env.company.id
+        domain = [("company_id", "=", company_id)]
+        if type_tax_use == "purchase":
+            domain.append(("type_tax_use", "=", "purchase"))
+        elif type_tax_use == "sale":
+            domain.append(("type_tax_use", "=", "sale"))
+        if price_include is False:
+            domain.append(("price_include", "=", False))
+        elif price_include is True:
+            domain.append(("price_include", "=", True))
+        # with the code above, if you set price_include=None, it will
+        # won't depend on the value of the price_include parameter
+        assert tax_dict.get("amount_type") in [
+            "fixed",
+            "percent",
+        ], "bad tax type, has to be fixed or percent"
+        assert "amount" in tax_dict, "Missing amount key in tax_dict"
+        domain.append(("amount_type", "=", tax_dict["amount_type"]))
+        if tax_dict.get("unece_type_code"):
+            domain.append(("unece_type_code", "=", tax_dict["unece_type_code"]))
+        if tax_dict.get("unece_categ_code"):
+            domain.append(("unece_categ_code", "=", tax_dict["unece_categ_code"]))
+        if tax_dict.get("unece_due_date_code"):
+            tax_exigibility = ato._get_tax_exigibility_from_unece_code(
+                tax_dict["unece_due_date_code"]
+            )
+            if tax_exigibility:
+                domain = expression.AND(
```

Comment:
Looks weird to use `AND` after appending domain elements all the time before (which is fine if you want `AND` operator and you control all the domain operands as is the case here)